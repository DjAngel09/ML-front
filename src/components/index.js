export * from './Bredcrumbs';
export * from './CardProduct';