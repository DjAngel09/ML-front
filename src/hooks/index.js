export * from './useFetchProducts';
export * from './useForm'; 
export * from './useFetchProduct';