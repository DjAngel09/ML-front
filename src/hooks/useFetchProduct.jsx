import { useEffect, useState } from 'react';
import { getProduct } from '../helpers';

export const useFetchProduct = (id) => {

    const defaultProduct = {
        item: {
            id: '',
            title: '',
            price: {
                currency: '',
                amount: 0
            },
            picture: '',
            condition: '',
            free_shipping: '',
            sold_quantity: '',
            description: ''
        }, 
        categories: []
    }

    const [product, setProduct] = useState(defaultProduct);
    const [isLoading, setIsLoading] = useState(true);

    useEffect(() => {

        const getData = async () => {
            const newProduct = await getProduct(id);
            setProduct(newProduct);
            setIsLoading(false);
        }
    
        getData();
    }, [id]);



    return {
        product,
        isLoading
    };

}