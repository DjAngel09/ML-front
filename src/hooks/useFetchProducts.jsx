import { useEffect, useState } from 'react';
import { getProducts } from '../helpers';

export const useFetchProducts = ( nameProduct ) => {
 
    const defaultProduct = {
        items: [],
        categories: []
    }

    const [products, setProducts] = useState(defaultProduct);
    const [isLoading, setIsLoading] = useState( true );
    
    useEffect( () => {

        const getData = async() => {
            setIsLoading(true);
            const newProducts = await getProducts ( nameProduct );
            setProducts(newProducts);
            setIsLoading(false);
        }

        getData();
    }, [nameProduct]);



    return {
        products,
        isLoading
    }

}