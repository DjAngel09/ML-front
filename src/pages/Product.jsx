import React from 'react';
import { useParams } from 'react-router-dom';
import { Bredcrumbs } from '../components';
import { useFetchProduct } from '../hooks';
export const Product = () => {

  const { id } = useParams();
  const { product, isLoading } = useFetchProduct(id);
  const { item, categories } = product
  return (
    <div>
      {
        isLoading
          ? (<h2>Cargando...</h2>)
          : <>
            <Bredcrumbs categories={categories} />
            <div className="card product-details p-16 animate__animated animate__fadeIn">
              <div className="row">
                <div className="col-12 col-lg-9">
                  <img src={item.picture} alt={item.title} />
                </div>
                <div className="col-12 col-lg-3">
                  <p className='fs-14 pt-16 pb-16' >{`${item.condition} - ${item.sold_quantity} vendidos`}</p>
                  <h1 className='fs-24 fw-bolder pb-32' >{item.title}</h1>
                  <p className='fs-46 pb-32' >$ {item.price.amount}</p>
                  <div className="d-grid gap-2 pr-32 pb-32">
                    <button className='btn btn-primary btn-block' >Comprar</button>
                  </div>
                </div>
                <div className="col-12 col-lg-9 pl-32">
                  <h3 className='fs-28 pb-32' >Descripci√≥n del producto</h3>
                  <p className='fs-16 pb-32' >{item.description}</p>
                </div>
              </div>
            </div>
          </>
      }
    </div>
  )
}