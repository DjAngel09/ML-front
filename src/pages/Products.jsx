import React from 'react';
import { useLocation } from 'react-router-dom';
import queryString from 'query-string';
import { useFetchProducts } from '../hooks';
import { CardProduct } from '../components/CardProduct';
import { Bredcrumbs } from '../components';

export const Products = () => {

  const location = useLocation();
  const { search = '' } = queryString.parse(location.search);

  const { products, isLoading } = useFetchProducts(search);

  return (
    <>
      {
        isLoading
          ? (<h2>Cargando...</h2>)
          : <>
            <Bredcrumbs categories={products.categories} />
            <div className='card mb-5' >

              {
                products.items.map(product => (
                  <div className='card-product p-16 animate__animated animate__fadeIn' key={product.id} role="heading">
                    <CardProduct product={product} />
                    <hr />
                  </div>
                ))
              }

            </div>
          </>
      }
    </>
  )
}
