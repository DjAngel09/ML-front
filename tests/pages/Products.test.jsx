import { render, screen } from "@testing-library/react";
import { Products } from "../../src/pages";
import { BrowserRouter as Router } from 'react-router-dom';
import { useFetchProducts } from "../../src/hooks";

jest.mock('../../src/hooks');


describe('testing in <Products />', () => {

    test('it should show the loading initially', () => {

        useFetchProducts.mockReturnValue({
            products: {
                items: [],
                categories: []
            },
            isLoading: true
        });

        render(
            <Router>
                <Products />
            </Router>
        );
        expect(screen.getByText('Cargando...'))
    });

    test('must show the products', () => {

        useFetchProducts.mockReturnValue({
            products: {
                items: [{
                    "id": "MLA1111948731",
                    "title": "Apple iPad Pro De 12.9  Wi-fi  256gb Gris Espacial (5ª Generación)",
                    "price": "355,699.00",
                    "picture": "http://http2.mlstatic.com/D_857823-MLA46975173456_082021-I.jpg",
                    "condition": "new",
                    "free_shipping": true,
                    "city": "Barracas"
                },
                {
                    "id": "MLA1130780887",
                    "title": "Apple iPad Pro De 12.9  Wi-fi  128gb Gris Espacial (5ª Generación)",
                    "price": "389,999.00",
                    "picture": "http://http2.mlstatic.com/D_689310-MLA46975173451_082021-I.jpg",
                    "condition": "new",
                    "free_shipping": true,
                    "city": "Balvanera"
                }],
                categories: [{
                    "id": "MLA1648",
                    "name": "Computación"
                }]
            },
            isLoading: false
        });

        const { container } = render(
            <Router>
                <Products />
            </Router>
        );
        expect( screen.getAllByRole('heading').length ).toBe(2);
        expect( container ).toMatchSnapshot();

    })

})