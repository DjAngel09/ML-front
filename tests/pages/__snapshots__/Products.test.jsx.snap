// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing in <Products /> must show the products 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="nav pt-16 pb-16 "
  >
    <ol
      class="breadcrumb"
    >
      <li
        class="breadcrumb-item"
      >
        Computación
      </li>
    </ol>
  </nav>
  <div
    class="card mb-5"
  >
    <div
      class="card-product p-16"
      role="heading"
    >
      <div
        class="row m-0"
      >
        <div
          class="col-12 col-md-2 ps-0 pr-16"
        >
          <img
            alt="Apple iPad Pro De 12.9  Wi-fi  256gb Gris Espacial (5ª Generación)"
            src="http://http2.mlstatic.com/D_857823-MLA46975173456_082021-I.jpg"
          />
        </div>
        <div
          class="col-12 col-md-8 ps-0 pe-0 pt-16 pb-16"
        >
          <p
            class="m-0 pb-32 fs-24 fw-normal"
          >
            $
            355,699.00
            <span
              class="icon-shipping ps-2"
            />
          </p>
          <p
            class="m-0 fs-18"
          >
            Apple iPad Pro De 12.9  Wi-fi  256gb Gris Espacial (5ª Generación)
          </p>
        </div>
        <div
          class="col-12 col-md-2 pt-16 pb-16"
        >
          <p
            class="m-0 fs-12"
          >
            Barracas
          </p>
        </div>
      </div>
      <hr />
    </div>
    <div
      class="card-product p-16"
      role="heading"
    >
      <div
        class="row m-0"
      >
        <div
          class="col-12 col-md-2 ps-0 pr-16"
        >
          <img
            alt="Apple iPad Pro De 12.9  Wi-fi  128gb Gris Espacial (5ª Generación)"
            src="http://http2.mlstatic.com/D_689310-MLA46975173451_082021-I.jpg"
          />
        </div>
        <div
          class="col-12 col-md-8 ps-0 pe-0 pt-16 pb-16"
        >
          <p
            class="m-0 pb-32 fs-24 fw-normal"
          >
            $
            389,999.00
            <span
              class="icon-shipping ps-2"
            />
          </p>
          <p
            class="m-0 fs-18"
          >
            Apple iPad Pro De 12.9  Wi-fi  128gb Gris Espacial (5ª Generación)
          </p>
        </div>
        <div
          class="col-12 col-md-2 pt-16 pb-16"
        >
          <p
            class="m-0 fs-12"
          >
            Balvanera
          </p>
        </div>
      </div>
      <hr />
    </div>
  </div>
</div>
`;
